samples/sensor/mcp9808/README.rst

```diff
@@ -6,42 +6,65 @@ MCP9808 Temperature Sensor
 Overview
 ********
 
-Sample application that periodically reads temperature from the MCP9808 sensor.
+This sample application periodically (0.5 Hz) measures the ambient
+temperature. The result is written to the console.
 
+If triggered measurements are enabled the sample initializes and
+maintains a 2 Cel window around the current temperature.  When the
+temperature moves outside the window an alert is given, and the window
+is moved to center on the new temperature.
 
 Requirements
 ************
 
 The MCP9808 digital temperature sensor converts temperatures between -20 |deg|
 C and +100 |deg| C to a digital word with |plusminus| 0.5 |deg| C (max.)
-accuracy. It is I2C compatible and
-supports up to 16 devices on the bus. We do not require pullup resistors on the
-data or clock signals as they are already installed on the breakout board.
-
-The MCP9808 is available in a discrete component form but it is much easier to
-use it mounted on a breakout board. We used the Adafruit breakout board.
-
-- `MCP9808 Sensor`_
-
-This sample uses the sensor APIs and the provided driver for the MCP9808 sensor.
+accuracy. It is I2C compatible and supports up to 16 devices on the bus.
 
 Wiring
 *******
 
-The MCP9808 requires 2 wires for the I2C bus plus power and ground. The power
-can be either 5V or 3.3V.
-
-We connect the Data and clock wires to Analog ports A4 and A5 which is the I2C
-pins on Arduino compatible boards.
-
-In this hookup we are only connecting one device to one of the supported boards.
-It reads the temperature and displays it on the console.
-
-
-References
-***********
-
-- http://www.microchip.com/wwwproducts/en/en556182
-
+The MCP9808 is available in a discrete component form but it is much easier to
+use it mounted on a breakout board.  We used the Adafruit `MCP9808
+Sensor`_ breakout board.
 
 .. _`MCP9808 Sensor`: https://www.adafruit.com/product/1782
+
+Building and Running
+********************
+
+After providing a devicetree overlay that specifies the sensor I2C bus
+and alert GPIO, build this sample app using:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/sensor/mcp9808
+   :board: particle_xenon
+   :goals: build flash
+
+Sample Output
+=============
+
+.. code-block:: console
+
+   *** Booting Zephyr OS build zephyr-v2.1.0-537-gbbdeaa1ae5bb  ***
+   Alert on temp outside [24500, 25500] mCel
+   Trtrigger fired 1, temp 15.9375 Cel
```

Comment:
These three lines should be
```
   trigger fired 1, temp 15.9375 Cel
   Alert on temp outside [15437, 16437] mCel
   Trigger set got 0
```

Reply:
@pabigot I see doc changes in #21547 that are different than these.  (Let's make sure the doc changes from this PR win.)

Reply:
I'll implement them in #21547; this PR is dependent on that one.

These three lines are exactly as produced by the sample, because the output is interleaved because two separate threads produce it.  I can add a comment explaining this, but it would be misleading to "fix" the output when it doesn't represent what the user will see.

Reply:
Sigh.  Yes, please explain the mangled output because separate threads are writing it (unless there's an API that prevents interleaving output available).